### PR TITLE
Replace `Simple` with `Scalar`

### LINKED ISSUE #942: [PEP] Rename "Simple" components to "Scalar"
## Proposal
Rename all "`Simple*`" components (`SimpleVar`, `SimpleBlock`, etc.) to "`Scalar*`" (`ScalarVar`, `ScalarBlock`, etc.).

## Motivation
There is nothing "simple" about the Pyomo Component API.  "Scalar" better represents what the "Simple" components are (it is a single thing), and more closely parallels the use of "Indexed" in the component hierarchy.

### PR DESCRIPTION
**Pull request by mrmundt**
## Summary/Motivation:
This partially resolves a PEP to rename `Simple` to `Scalar` as `Simple` is a misnomer.

- [x] Add deprecations on all changes from `Simple` @jsiirola 

## Changes proposed in this PR:
- Rename `Simple` to `Scalar`
- Update tests to reflect new names
- Fix some typos

### Legal Acknowledgement

By contributing to this software project, I have read the [contribution guide](https://pyomo.readthedocs.io/en/stable/contribution_guide.html) and agree to the following terms and conditions for my contribution:

1. I agree my contributions are submitted under the BSD license.
2. I represent I am authorized to make the contributions and grant the license. If my employer has rights to intellectual property that includes these contributions, I represent that I have received permission to make contributions and grant the required license on behalf of that employer.

### CODE REVIEW

File: pyomo/core/expr/visitor.py [hunk 1]
```diff
@@ -342,15 +342,15 @@ def walk_expression(self, expr):
                         ptr[2].__exit__(None, None, None)
                 ptr = ptr[0]
 
-class SimpleExpressionVisitor(object):
+class ScalarExpressionVisitor(object):
```
Comment (jsiirola) [inline pyomo/core/expr/visitor.py:345]: This should probably be left "Simple"

File: pyomo/core/tests/unit/test_action.py [hunk 2]
```diff
@@ -63,6 +63,8 @@ def test_getattr(self):
         self.assertEqual( self.instance.A.value, 4.3)
 
 
+Simple = Scalar
```
Comment (jsiirola) [inline pyomo/core/tests/unit/test_action.py:66]: This is a test suite, I don't think we need the "backwards compatibility" alias.

File: pyomo/core/tests/unit/test_check.py [hunk 3]
```diff
@@ -89,7 +89,7 @@ def test_false(self):
         except ValueError:
             pass
 
-
+Simple = Scalar
```
Comment (jsiirola) [inline pyomo/core/tests/unit/test_check.py:92]: This is a test suite, I don't think we need the "backwards compatibility" alias.

**Review by jsiirola — DISMISSED** [review]
A couple minor changes.  I will work on the "RenamedClass" metaclass shortly.

**Comment by blnicho** [comment]
@jsiirola looks like the failing tests are all coming from Pyomo connectors which was deprecated in 5.6.9. Any objection to removing connectors entirely?

**Comment by blnicho** [comment]
@jsiirola I just noticed a bunch of `FooSimpleSet` classes in `pyomo/core/base/set.py`, I assume we want to rename these as well?

**Comment by mrmundt** [comment]
@jsiirola @blnicho - Thanks, Bethany, for finishing this up!

**Comment by blnicho** [comment]
I think there are a few more `Simple` components to change in `set.py` and then this PR should be ready.

**Review by blnicho — APPROVED** [review]

**Pull request merged**